# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmFrameIter.h
# Commit: db39bf2058e1
# Full Hash: db39bf2058e13e89ac9238aa864df13a1c41a163
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-10-20 14:55:47
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 4: Untie frame iteration from Frame::tls. r=lth
#   
#   Here we replace usage of Frame::tls in frame iteration with GetNearestEffectiveTls.
#   We also maintain current tls for frame iteration object to not to call GetNearestEffectiveTls everytime.
#   
# ==============================================================================

diff -r a072d5ee2511 -r db39bf2058e1 js/src/wasm/WasmFrameIter.h
--- a/js/src/wasm/WasmFrameIter.h	Tue Oct 20 10:24:32 2020 +0000
+++ b/js/src/wasm/WasmFrameIter.h	Tue Oct 20 10:24:40 2020 +0000
@@ -65,6 +65,7 @@
   const CodeRange* codeRange_;
   unsigned lineOrBytecode_;
   Frame* fp_;
+  TlsData* tls_;
   uint8_t* unwoundIonCallerFP_;
   jit::FrameType unwoundIonFrameType_;
   Unwind unwind_;
@@ -95,6 +96,7 @@
   jit::FrameType unwoundIonFrameType() const;
   uint8_t* unwoundIonCallerFP() const { return unwoundIonCallerFP_; }
   Frame* frame() const { return fp_; }
+  TlsData* tls() const { return tls_; }
 
   // Returns the address of the next instruction that will execute in this
   // frame, once control returns to this frame.
